airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -241,7 +242,9 @@ def _process_log(self, itr):
         """
         # Consume the iterator
         for line in itr:
-            line = line.decode('utf-8').strip()
+            if isinstance(line, six.binary_type):
```

Comment:
Why don't we use "universal_newlines" as part of the Popen, wouldn't that solve the issue without this check? (see also airflow.security.kerberos.